### ISSUE TITLE
Don't add notes when the question is invalid

### ISSUE THREAD

**Issue by SomeoneElseOSM** [westnordost/StreetComplete#351]
A lot of StreetComplete notes in the UK (scanned by looking through #osm-gb) are basically "not an answer to the question, because the question was invalid".

Example:
https://www.openstreetmap.org/note/1027000

Also asking for road names for car parks and turning circles etc. makes no sense.

Make it clear when you are asking for e.g. a house  number that you want a house number, not a name, or a description, or the colour of the owner's cat, and allow people to say "this question doesn't make sense" (providing you with feedback about what questions work) and not adding an OSM note which will confuse everyone else.

**Comment by westnordost** [comment]
> Don't add notes when the question is invalid

Eh? No. This is exactly the reason to add notes, because if a question is invalid, it is very probable that the underlying data is incorrect.

I cleared up the wording (already), see the other issue from you that I commented and closed. Now it should be clear that the note was only written because the user was not able to answer the question. The usual reason why a user is not able to answer a question is because the question was invalid (=based on wrong data).
Naturally, the app cannot detect wrong data, only the surveyor can. And that is why for every question type, it is possible for the surveyor to leave a note.

**Also**:
* The app does not ask road names for car parks and turning circles.
* The app makes clear that it is asking for a house number and not the color of the owner's cat.

If you find that the app is asking wrong questions even though the underyling data is correct, then you are free to report this. The prerequisite of this is however that your allegations are based on correct observations.

**Comment by SomeoneElseOSM** [comment]
No, based on the notes added in the UK in the last couple of days an invalid question does not mean wrong data, but a series of incorrect assumptions about real-world data on behalf of the app author.  Have a look at the notes feed if you don't believe me.

Taking https://www.openstreetmap.org/note/1027000 again as an example, it is of no benefit to anyone to add that as a note.  

Not every building is a house, and not every building has a house number.  If you don't control how sensible the notes that this app adds then people will just think "that's a StreetComplete note, it's likely to be rubbish, so I'll just close it".  I'm not saying whether that's right or wrong, but it will happen unless you take more responsibility for the notes that you are adding.

**Comment by krzyk** [comment]
The app does not ask for housenumbers for all buildings, the list you can find here: https://github.com/westnordost/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/housenumber/AddHousenumber.java#L33-L36

Basically it has quite a conservative list to avoid asking for housenumbers for buildings that won't have it.
In your example it asked for the housenumbers because the building is marked as university.

I'm not sure how to avoid that (except by excluding `building=university`). In my country for university buildings we add housename so I don't have that trouble with StreetComplete (because it excludes such buildings), maybe there should be somekind of exclusion based on name + university zone?

Do you have other examples of wrongly added notes? Right now we can see that universities (or that particular university) in UK are wrongly handled.

**Comment by westnordost** [comment]
Okay, lets make this issue into this specific problem and create another issue for each "oddly asked question", discuss, and solve.

**Comment by westnordost** [comment]
So, the question is: Can it be assumed that universities (, schools, kindergartens,...) generally have no housenumber? Then, I can simply remove this building type from the white list.

The reality however may be different. I have the impression that it can only be assumed with relative certainty that _any_ buildings within an amenity (at least school, kindergarten, university, perhaps more) do not have housenumbers _as long as there is a housenumber on the whole amenity already_.
The app does currently not check for that, but it could (with a little more effort). Would it make sense to check that? And if it would, for what other areas should it check that may "override" the housenumbers? Perhaps commercial and industrial landuses?
I know that at least or landuses, it is discouraged to set the housenumber on the whole area instead of the building with the post box/main entrance but I reckon that this may be the way it is mapped now or make more sense for specific cases.

Opinions?

**Comment by westnordost** [comment]
@krzyk Or perhaps also exclude buildings with names?

**Comment by westnordost** [comment]
Hm actually, even if I implemented the things I mentioned in my comment yesterday, it wouldn't fix it for the example note you gave because the university campus as a whole also does not have a housenumber.

Two possible solutions come to my mind:
1. only show quests for universities, schools, kindergartens,.... when the amenity is tagged on the building, meaning the amenity is only in this building.
2. Offer the user to (instead of leaving a note) answer that the building indeed has no address while acknowledging that the tagged building type is correct. (i.e. not a garage)

**Comment by matkoniecz** [comment]
> Can it be assumed that universities (, schools, kindergartens,...) generally have no housenumber?

amenity=university object is very unlikely to have an address in Poland. Typical university will include multiple buildings, each with separate address.

I would expect that building=university would have an address but there are probably cases where this will not be true (quite rare, maybe very rare).

Schools, kindergartens are more likely to be limited to a single building/building complex with a single address.

**Comment by westnordost** [comment]
There is the (so far rarely used) tag https://wiki.openstreetmap.org/wiki/Key:noaddress for the purpose of tagging that a building indeed has no name.

However, the more likely case that a building has no housenumber is that it is wrongly tagged as a i.e. house even though it is a garage etc.

Implementation should be like this: 

User can answer "has no housenumber". The app will then pop up a confirmation dialog, asking whether that building is really a XXXXXX. If the user answers "no", the user is prompted to leave a note to explain what kind of building it is instead or even better directly re-tag the building. If the user answers "yes", then and only then it is tagged with noaddress=yes.

This implementation means that it is blocked by the "building type" quest to be implemented.

**Comment by SomeoneElseOSM** [comment]
That doesn't address the root of the problem here - the user doesn't get a chance to say that "everything in OSM is just fine" and NOT leave a note and NOT add a rarely used tag that few consumers will anticipate.

**Comment by westnordost** [comment]
Yes, that is a general problem in OSM. There is no general distinction between "a property does not exist on an object" and "a property hasn't been determined on an object". But, this is information that is simply necessary to do proper detailed survey for missing data. StreetComplete is only a tool to make it easier to do this kind of survey, the root problem would be just the same without this app.

This is why tags like noaddress=yes, noname=yes etc exist.

However, I do not see any problem with using a "rarely used tag". The fact that few consumers will process it is not any argument for or against any tag. It is not as if any consumer will refuse to work because there is some (for the application) unknown tag on an object.

**Comment by westnordost** [comment]
Ah right, it's blocked. To recap what needs to be done here:

First, wait for https://github.com/westnordost/StreetComplete/pull/774 to be complete, to have the translations for the common building type names.

Then, add _"has no housenumber"_ as answer-option. This opens a confirmation dialog:
  **This building is captured as: ... . Is this correct but it simply does not have an own housenumber?
  [No (Leave Note)] [Yes]**
Clicking No, leaves a note, clicking yes, adds `noaddress=yes` to the object.

**Comment by exploide** [comment]
> Clicking No, lets the user leave a note and explain what it is then

When this is blocked by the building type quest, why wouldn't "*has no housenumber*" option hand over to the building type quest to specify a better value for `building=*` instead of leaving a note?

**Comment by muralito** [comment]
Please do not ask for housenumber of a building when the building has multiple entrances. In these case every address should be in every entrance point.

**Comment by matkoniecz** [comment]
Please create a new issue for new issues rather than comment on old closed ones.

It is both more effective and less confusing.

Also, on creating a new issue please mention your country/area.
